src/Mod/CAM/Path/Main/Gui/SimulatorGL.py

```diff
@@ -0,0 +1,309 @@
+# -*- coding: utf-8 -*-
+# ***************************************************************************
+# *   Copyright (c) 2017 Shai Seger <shaise at gmail>                       *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+import FreeCAD
+import Path
+import Path.Base.Util as PathUtil
+import Path.Dressup.Utils as PathDressup
+import PathScripts.PathUtils as PathUtils
+import Path.Main.Job as PathJob
+import PathGui
+import CAMSimulator
+import math
+import os
+
+from FreeCAD import Vector, Base
+from PySide2.QtWidgets import QDialogButtonBox
```

Comment:
This is only going to work with Qt5/PySide2, it would be better to use `PySide.QtWidgets`, also this can go under the `FreeCAD.GuiUp` check.